tests/components/unifi/test_init.py

```diff
@@ -140,33 +90,8 @@
 
 async def test_unload_entry(hass):
     """Test being able to unload an entry."""
-    entry = MockConfigEntry(
-        domain=unifi.DOMAIN,
-        data={
-            "controller": {
-                "host": "0.0.0.0",
-                "username": "user",
-                "password": "pass",
-                "port": 80,
-                "site": "default",
-                "verify_ssl": True,
-            },
-            "poe_control": True,
-        },
-    )
-    entry.add_to_hass(hass)
+    controller = await setup_unifi_integration(hass)
+    assert hass.data[unifi.DOMAIN]
 
-    with patch.object(unifi, "UniFiController") as mock_controller, patch(
-        "homeassistant.helpers.device_registry.async_get_registry",
-        return_value=mock_coro(Mock()),
-    ):
-        mock_controller.return_value.async_setup.return_value = mock_coro(True)
-        mock_controller.return_value.mac = "00:11:22:33:44:55"
-        assert await unifi.async_setup_entry(hass, entry) is True
-
-    assert len(mock_controller.return_value.mock_calls) == 1
-
-    mock_controller.return_value.async_reset.return_value = mock_coro(True)
-    assert await unifi.async_unload_entry(hass, entry)
-    assert len(mock_controller.return_value.async_reset.mock_calls) == 1
-    assert hass.data[unifi.DOMAIN] == {}
+    assert await unifi.async_unload_entry(hass, controller.config_entry)
+    assert not hass.data[unifi.DOMAIN]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4925854)